### ISSUE TITLE
DGL Baselines for MoleculeNet

### ISSUE THREAD

**Issue by mufeili** [deepchem/deepchem#2233]
During the meeting on [9/25/2020](https://forum.deepchem.io/t/deepchem-minutes-9-25-2020/271), @rbharath and I both liked the idea of integrating DGL-based baselines for MoleculeNet with DeepChem. This can serve the following purposes:
- Establish standard baselines for newcomers
- Standardize the baseline submission process and a fair evaluation setting
- Drive DGL integration

My ongoing effort for MoleculeNet baselines is in [this PR for DGL-LifeSci](https://github.com/awslabs/dgl-lifesci/pull/104). I'm close to finish 10-14 baselines for 12 datasets -- BACE, BBBP, ClinTox, FreeSolv, HIV, Lipophilicity, MUV, PCBA, SIDER, Tox21, ToxCast, ESOL.

For all these datasets, I split them into train/val/test subsets using scaffold split with a ratio of 0.8:0.1:0.1. I used Bayesian optimization for searching the hyperparameters based on validation performance.

For each baseline on each dataset, I provide both the training script, the hyperparameters, and a pre-trained model for reproducing the result since I did not fix the random seed during the training.

For the next steps:
1. What will be the main page for all baseline submissions? This should be something like [the OGB benchmark here](https://ogb.stanford.edu/docs/leader_nodeprop/).
2. Where should we place the scripts for training from scratch/evaluating pre-trained models? I can also develop model wrappers along the way as @nd-02110114 proposed previously in #2131 .

**Comment by rbharath** [comment]
This is awesome work @mufeili!

Here are some first thoughts on how we can start building in this infrastructure. Also, cc'ing @miaecle who might be interested

> My ongoing effort for MoleculeNet baselines is in [this PR for DGL-LifeSci](https://github.com/awslabs/dgl-lifesci/pull/104). I'm close to finish 10-14 baselines for 12 datasets -- BACE, BBBP, ClinTox, FreeSolv, HIV, Lipophilicity, MUV, PCBA, SIDER, Tox21, ToxCast, ESOL.

This is awesome! We should also extend this outwards to the new V2 MoleculeNet datasets (see the new tutorial https://github.com/deepchem/deepchem/blob/master/examples/tutorials/03_An_Introduction_To_MoleculeNet.ipynb). There's something like 30 datasets now including V2. Not all the V2 datasets are molecular datasets, but we should run benchmarks on those that are.

> 1. What will be the main page for all baseline submissions? This should be something like [the OGB benchmark here](https://ogb.stanford.edu/docs/leader_nodeprop/).

A couple of options come to mind. We could add a page to the deepchem.io website or to the DeepChem readthedocs to add a MoleculeNet leaderboard. We could also update the moleculenet.ai site itself, but we don't currently have the source for that in version control (@miaecle might be able to help on this though)

> 2. Where should we place the scripts for training from scratch/evaluating pre-trained models? I can also develop model wrappers along the way as @nd-02110114 proposed previously in #2131 .

How about the https://github.com/deepchem/moleculenet repo for the scripts? This is mostly empty right now, but we can use it as the staging ground for reproduction scripts. The model wrappers should go in the main DeepChem repo.

For the submission process of a new model to MoleculeNet how about something as follows?

1. Make a PR to https://github.com/deepchem/moleculenet that adds a new directory with your scripts for reproduction of your results. This directory should include
    - Weights for your trained models with scripts to run these trained models and reproduce results
    - Scripts that enable reproduction of your hyperparameter search process
2. (Optional, but recommended) If your models are implemented in Python, contribute a wrapper/implementation of the core model to the main DeepChem repo
3. A DeepChem developer will run your scripts independently on a new machine to verify results. Both the trained weights and the hyperparameter search script will be run. If your model is very compute intensive, DeepChem developers may ask for your help in donating credits/funds to run the job.
4. Once results are replicated, the leaderboard will be updated with the new results

This is a little manual but I figure simple might be better to start. What do folks think? 

CC @peastman @ncfrey @nd-02110114  who might also be interested

**Comment by mufeili** [comment]
> This is awesome! We should also extend this outwards to the new V2 MoleculeNet datasets (see the new tutorial https://github.com/deepchem/deepchem/blob/master/examples/tutorials/03_An_Introduction_To_MoleculeNet.ipynb). There's something like 30 datasets now including V2. Not all the V2 datasets are molecular datasets, but we should run benchmarks on those that are.

Great to know that! For a proof of concept on establishing the benchmark, I think we can first figure out all the modeling & submission details based on the 12 datasets. We can pick up the rest datasets after that.

> A couple of options come to mind. We could add a page to the deepchem.io website or to the DeepChem readthedocs to add a MoleculeNet leaderboard. We could also update the moleculenet.ai site itself, but we don't currently have the source for that in version control (@miaecle might be able to help on this though)
>
> How about the https://github.com/deepchem/moleculenet repo for the scripts? This is mostly empty right now, but we can use it as the staging ground for reproduction scripts. The model wrappers should go in the main DeepChem repo.

These all sound very reasonable.

> For the submission process of a new model to MoleculeNet how about something as follows?

> Make a PR to https://github.com/deepchem/moleculenet that adds a new directory with your scripts for reproduction of your results. This directory should include
Weights for your trained models with scripts to run these trained models and reproduce results
Scripts that enable reproduction of your hyperparameter search process
(Optional, but recommended) If your models are implemented in Python, contribute a wrapper/implementation of the core model to the main DeepChem repo
A DeepChem developer will run your scripts independently on a new machine to verify results. Both the trained weights and the hyperparameter search script will be run. If your model is very compute intensive, DeepChem developers may ask for your help in donating credits/funds to run the job.
Once results are replicated, the leaderboard will be updated with the new results

For the DGL baselines, I would recommend first contributing the model wrappers to DeepChem core and then the scripts to the moleculenet repo as common baselines are used for multiple benchmarks. For a general submission, I think it can also be optional for submitting scripts to https://github.com/deepchem/moleculenet. It should be fine as long as people have scripts/weights publicly available somewhere, say in their own repo and the leaderboard has a link for each submission. There are two reasons: 1) many people might develop submissions in conducting their own research and they may have a github repo associated with their research 2) people who make submissions should be responsible for maintaining their submission code. Having all submission scripts in the moleculenet repo might bring too much burden to the DeepChem team for maintenance.

> If your model is very compute intensive, DeepChem developers may ask for your help in donating credits/funds to run the job.

It will be better if there is a sponsor for this.
